model-optimizer/extensions/front/tf/TFSliceToSlice.py

```diff
@@ -43,8 +44,13 @@ def replace_sub_graph(self, graph: Graph, match: dict):
         rename_nodes([(node, slice_name + '/to_be_removed'), (slice_node, slice_name)])
 
         eq_node = Equal(graph, {'name': slice_name + '/equal'}).create_node()
-        minus_one_node = Const(graph, {'name': slice_name + '/minus_one', 'value': np.array(-1)}).create_node()
-        int32_max_node = Const(graph, {'name': slice_name + '/int32_max', 'value': np.iinfo(np.int32).max}).create_node()
+        minus_one_node = Const(graph, {'name': slice_name + '/minus_one', 'value': int64_array(-1)}).create_node()     
+        
+        #Select requires equal dtypes on ports 1 and 2.
+        #Dtype of [2] port is int64(summ_node -> cast to int64),
+        #so dtype of [1] (int32_max_node)  must be the same
+        int32_max_node = Const(graph, {'name': slice_name + '/int32_max', 
+                        'value': int64_array(np.iinfo(np.int32).max)}).create_node()
```

Comment:
@lazarevevgeny, the transformation looks very tricky and relies on some implementation specifics from middle phase (`create_ss_interval_border`) and support of out of bound values for end by `StridedSlice`. I propose to use different transformation that is more evident and does not use tricky logic with max values.
Transformation idea:
1. Compute `where_minus_one = Equal (size, minus_one)`
2. Compute `max_ends = ShapeOf(Data)`
3. Compute partial correct ends `p_ends = Add(begin, sizes)`
4. Compute final end values `ends= Select(where_minus_one, max_ends, p_ends)`

Reply:
@rkazants, thanks for your comment! But I disagree that this transformation relies on `create_ss_interval_border` instead it already inserts int32.max ends. 

You propose a more clean solution. But I am not sure that we will get functional advantages if we rewrite this transformation according to your proposal, it appears to me that implementation cost does not worth the benefits we get.

Reply:
1. The transformation depends on `create_ss_interval_border ` - False.
2. The transformation depends on support of out of bound values for end by `StridedSlice` - True.
3. Tricky logic with max values - True.
4. The proposed solution more clean and robust (it will work for cases when input tensors are huge (dimension is more than max<int32>) which, however, we have not seen yet) - True.

In addition:

5. This is a front phase transformation so we can implement it shorter without usage of connections and ports in most cases - True.

Many places in this transformation code can be written like this:
```
select_node = Select(graph, {'name': slice_name + '/select'}).create_node([eq_node, int32_max_node, cast])
```

instead of the following lines:
```
select_node.in_port(2).get_connection().insert_node(cast)
eq_node.out_port(0).connect(select_node.in_port(0))
int32_max_node.out_port(0).connect(select_node.in_port(1))
```

**Decision: rewrite the transformation taking into account the text above.**